### DISCUSSION TITLE
time.py

### DISCUSSION THREAD

**Discussion by angelminsua** [mesa/mesa#1314] [Q&A] [answered]
_agents defined in BaseScheduler class (time.py) is a OrderedDict() 
and the method  agents(self) defined in BaseScheduler class returns   _agents 
                                                             @property
                                                              def agents(self) -> OrderedDict:
                                                                  return (self._agents)

Why when I print the type of schedule.agents in bank_reserves example puts <class 'list'>  ?

**Comment by rht** [comment, marked as answer]
I'm not sure which version of the code is that, but if you look at the latest main, it is fine:
https://github.com/projectmesa/mesa/blob/a0e5d100c9992806ee28f2cc8923b35c8f0197d3/mesa/time.py#L90-L92

**Reply by angelminsua** [reply]
Ok. I was using an old version. 
In the lastest version you can't get a specific agent by id. This is a problem when you need to get it.
